minio/lib/minio/config.rb

```diff
@@ -0,0 +1,19 @@
+# frozen_string_literal: true
+
+module MinioRuby
+  class Config
+    attr_accessor :endpoint, :port, :transport, :secure
+    attr_accessor :access_key, :secret_key, :region, :service
+
+    def initialize(args = {})
+      @endpoint   = args[:endpoint] || 'http://localhost:9000'
```

Comment:
the endpoint shouldn't be of form `http://<hostname>:<port>` you should take endpoint of form `<hostname>:<port>` and avoid taking `<port>` as a special argument.